inspirehep/modules/xrootd/ext.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
```

Comment:
The script is idempotent, but what is produced is not guaranteed to be correct, as in the example above. If you read its code it's just a simple text substitution that takes the first 21 lines that look like comments and replaces those with the right license. This was enough for the typical case, in which a license was wrong because it had the wrong year or referred to the wrong project.

It's easy to make it more robust (better algorithm: you remove lines until you see something that is not a comment, and then you add the right license back), but I'd rather not fail builds because of a wrong license or wrongly formatted files: it's just frustrating for the developer (Trust me, I've been there with Invenio 2). I'd like to add this as a smudge filter, so that all this is both correct and invisible to the developer.